tiledb/sm/query/sparse_global_order_reader.cc

```diff
@@ -1061,6 +1057,70 @@ Status SparseGlobalOrderReader::copy_fixed_data_tiles(
   return Status::Ok();
 }
 
+Status SparseGlobalOrderReader::copy_timestamps_tiles(
+    const uint64_t num_range_threads,
+    const std::vector<ResultCellSlab>& result_cell_slabs,
+    const std::vector<uint64_t>& cell_offsets,
+    QueryBuffer& query_buffer) {
+  auto timer_se = stats_->start_timer("copy_timestamps_tiles");
+
+  // Process all tiles/cells in parallel.
+  auto status = parallel_for_2d(
+      storage_manager_->compute_tp(),
+      0,
+      result_cell_slabs.size(),
+      0,
+      num_range_threads,
+      [&](uint64_t i, uint64_t range_thread_idx) {
+        // For easy reference.
+        auto& rcs = result_cell_slabs[i];
+        auto rt = (ResultTileWithBitmap<uint8_t>*)result_cell_slabs[i].tile_;
+        const uint64_t cell_size = sizeof(uint64_t);
+
+        // Get source buffers.
+        const auto tile_tuple = rt->tile_tuple(constants::timestamps);
+        const auto t = &std::get<0>(*tile_tuple);
+
+        // Compute parallelization parameters.
+        auto&& [min_pos, max_pos, dest_cell_offset, skip_copy] =
+            compute_parallelization_parameters(
+                range_thread_idx,
+                num_range_threads,
+                rcs.start_,
+                rcs.length_,
+                cell_offsets[i]);
+        if (skip_copy) {
+          return Status::Ok();
+        }
+
+        // Get dest buffer.
+        auto buffer =
+            static_cast<uint64_t*>(query_buffer.buffer_) + dest_cell_offset;
```

Comment:
This shouldn't be `dest_cell_offset * cell_size` ?

Reply:
No, since it is cast as a uint64_t*, cell_size is already implied.